### ISSUE TITLE
Interesting but unusable. Too much false positive.

### ISSUE THREAD

**Issue by Vadim2S** [bcmi/DIRL#5]
I am try DIRL with my own little test dataset. In 48 total here is only 2 success and around 45 false positive alarms.

For example: (fake, mask, DIRL result)
![military118_1_1](https://user-images.githubusercontent.com/84853762/159417496-4aaec92c-7f57-4521-b7fb-1009bd94c539.jpg)

**Comment by jimleungjing** [comment]
> I am try DIRL with my own little test dataset. In 48 total here is only 2 success and around 45 false positive alarms.
> 
> For example: (fake, mask, DIRL result) ![military118_1_1](https://user-images.githubusercontent.com/84853762/159417496-4aaec92c-7f57-4521-b7fb-1009bd94c539.jpg)

Our model is trained on iHarmony4 dataset. It is better to finetune our model on your own dataset.

**Comment by alishan2040** [comment]
@Vadim2S yes fine-tuning is a good idea. 
@jimleungjing Did you train your network with different backbone (resnet50 instead of resnet34)? There is an option in code to change backbone but the code did not work when we do so. Is it possible if you can review the problem with changing backbone from resnet34 to resnet50?
**ERROR:**
weight of size [256, 256, 3, 3], expected input[8, 512, 226, 226] to have 256 channels, but got 512 channels instead. 

I understand this is an issue with input dimensions but with the given code, it should work because dimensions of resnet50 are same in both encoder and decoder part.

![image](https://user-images.githubusercontent.com/30209002/164038854-ddd29e0b-a0f2-4249-9fff-3754b98ff0cf.png)
![image](https://user-images.githubusercontent.com/30209002/164038706-a7e8fbba-5249-4ba9-b9c2-ea77318866a3.png)

Thanks!!!

**Comment by jimleungjing** [comment]
> @Vadim2S yes fine-tuning is a good idea. @jimleungjing Did you train your network with different backbone (resnet50 instead of resnet34)? There is an option in code to change backbone but the code did not work when we do so. Is it possible if you can review the problem with changing backbone from resnet34 to resnet50? **ERROR:** weight of size [256, 256, 3, 3], expected input[8, 512, 226, 226] to have 256 channels, but got 512 channels instead.
> 
> I understand this is an issue with input dimensions but with the given code, it should work because dimensions of resnet50 are same in both encoder and decoder part.
> 
> ![image](https://user-images.githubusercontent.com/30209002/164038854-ddd29e0b-a0f2-4249-9fff-3754b98ff0cf.png) ![image](https://user-images.githubusercontent.com/30209002/164038706-a7e8fbba-5249-4ba9-b9c2-ea77318866a3.png)
> 
> Thanks!!!

I just fixed this bug. However, I suggest that you should reduce the dimension of encoder outputs since the output dimension of resnet50 will lead to super heavy model and take more GPU memory to train. For example, you can apply an extra conv1x1 to reduce the encoder tensor dimension.

**Comment by alishan2040** [comment]
> > @Vadim2S yes fine-tuning is a good idea. @jimleungjing Did you train your network with different backbone (resnet50 instead of resnet34)? There is an option in code to change backbone but the code did not work when we do so. Is it possible if you can review the problem with changing backbone from resnet34 to resnet50? **ERROR:** weight of size [256, 256, 3, 3], expected input[8, 512, 226, 226] to have 256 channels, but got 512 channels instead.
> > I understand this is an issue with input dimensions but with the given code, it should work because dimensions of resnet50 are same in both encoder and decoder part.
> > ![image](https://user-images.githubusercontent.com/30209002/164038854-ddd29e0b-a0f2-4249-9fff-3754b98ff0cf.png) ![image](https://user-images.githubusercontent.com/30209002/164038706-a7e8fbba-5249-4ba9-b9c2-ea77318866a3.png)
> > Thanks!!!
> 
> I just fixed this bug. However, I suggest that you should reduce the dimension of encoder outputs since the output dimension of resnet50 will lead to super heavy model and take more GPU memory to train. For example, you can apply an extra conv1x1 to reduce the encoder tensor dimension.

@jimleungjing Thanks for fixing it. Yes you are right, changing the backbone to ResNet50 results in more memory cost. Could you please guide if I need to add conv1x1 after each encoder layer (output). Wouldn't that change the dimensions in the decoder too? What changes do I have to make for code to work? Also if you've pretrained weights with backbone 50 on iHormany4 datasets, please share?

Thanks again!

**Comment by jimleungjing** [comment]
Well, you can add conv1x1 in the decoder to reduce tensor dimension, and rewrite ```self.dims``` according to the output dimension of conv1x1, other code should remain the same. You can check out the tensor dimension in debug mode.

**Comment by alishan2040** [comment]
Okay. As I noticed fine-tuning works better than training from scratch. Did you train your model on iHarmony4 dataset with backbone Resnet50?

**Comment by jimleungjing** [comment]
> Okay. As I noticed fine-tuning works better than training from scratch. Did you train your model on iHarmony4 dataset with backbone Resnet50?

No, you can try it.
